packages/frontend/editor-ui/src/stores/users.store.ts

```diff
@@ -42,27 +36,24 @@ const _isInstanceOwner = (user: IUserResponse | null) => user?.role === ROLE.Own
 const _isDefaultUser = (user: IUserResponse | null) =>
 	_isInstanceOwner(user) && _isPendingUser(user);
 
+type LoginHook = (user: CurrentUserResponse) => void | Promise<void>;
+type LogoutHook = () => void | Promise<void>;
```

Comment:
1. None of them are async, so we could simplify the signatures.
2. `LoginHook` maybe shouldn't be coupled to `CurrentUserResponse` but rather only know about the components of the user that it cares about? Not sure here.

Reply:
Thanks for the feedback. I've removed the async nature of it. I'll leave it bound to `CurrentUserResponse` for now. We can narrow it down when we move the store out.